components/scream/src/physics/spa/spa_functions_impl.hpp

```diff
@@ -300,32 +301,30 @@ perform_vertical_interpolation(
 template <typename S, typename D>
 void SPAFunctions<S,D>
 ::set_remap_weights_one_to_one(
-    const Int                ncols_scream,
     gid_type                 min_dof,
     const view_1d<gid_type>& dofs_gids,
           SPAHorizInterp&    spa_horiz_interp
   )
 {
   // There may be cases where the SPA data is defined on the same grid as the simulation
   // and thus no remapping is required.  This simple routine establishes a 1-1 horizontal
-  // mapping.
+  // mapping
   auto num_local_cols = dofs_gids.size();
-  spa_horiz_interp.length            = num_local_cols; 
-  spa_horiz_interp.source_grid_ncols = ncols_scream;
-  spa_horiz_interp.weights           = view_1d<Real>("",spa_horiz_interp.length);
-  spa_horiz_interp.source_grid_loc   = view_1d<gid_type> ("",spa_horiz_interp.length);
-  spa_horiz_interp.target_grid_loc   = view_1d<gid_type> ("",spa_horiz_interp.length);
-  Kokkos::deep_copy(spa_horiz_interp.weights,1.0);
-  Kokkos::parallel_for("", num_local_cols, KOKKOS_LAMBDA(const int& ii) {
-    spa_horiz_interp.target_grid_loc(ii) = ii;
-    // Note we are interested in the vector index, not the actual global-id 
-    // Here we want the index in a the source data vector corresponding to this column
-    // which needs to be offset by the minimum degree of freedom in the whole grid.
-    // That way the first global-id will map to the 0th entry in the source grid data.
-    spa_horiz_interp.source_grid_loc(ii) = dofs_gids(ii) - min_dof;
-  });
-  // Determine the set of unique columns in this remapping
-  spa_horiz_interp.set_unique_cols();
+  auto& spa_gsmap = spa_horiz_interp.gsmap;
+  spa_gsmap = GSMap(spa_horiz_interp.m_comm,"SPA 1-1 Remap",dofs_gids,min_dof);
+  view_1d<gid_type> src_dofs("",1);
+  view_1d<Real>     src_wgts("",1);
+  auto dofs_gids_h = Kokkos::create_mirror_view(dofs_gids);
+  Kokkos::deep_copy(dofs_gids_h,dofs_gids);
+  for (int ii=0;ii<num_local_cols;ii++) {
```

Comment:
This is done once at model init, right?

Reply:
yes, just once, for cases where there is no remap.  Such as in an ne30np4 case where the data is already on the native grid.

Reply:
Worth documenting in the comment above the function?